### PR TITLE
WFCORE-5263 Update all projects with a dependency on wildfly-elytron to use individual modules

### PR DESCRIPTION
**Pull request by parsharma**
Issue: https://issues.redhat.com/browse/WFCORE-5263

### CODE REVIEW

**Comment by parsharma** [comment]
Still work is going on. I have raised this PR before the whole changes to test if anything is breaking because of it.

**Comment by jmesnil** [comment]
@parsharma I'll put this PR on hold to avoid merging until it is ready.

Please add a comment when you need a review.

**Comment by parsharma** [comment]
@jmesnil the work is done. PR is ready for review. Thanks

**Comment by jmesnil** [comment]
@fjuma or @darranl could you please review?

File: testsuite/pom.xml [hunk 1]
```diff
@@ -212,6 +212,18 @@
...
+            <groupId>org.wildfly.security</groupId>
+            <artifactId>wildfly-elytron-auth</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.wildfly.security</groupId>
+            <artifactId>wildfly-elytron-auth-server</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.wildfly.security</groupId>
+            <artifactId>wildfly-elytron</artifactId>
```
Comment (fjuma) [inline testsuite/pom.xml:225]: This should be removed.

**Review by fjuma — CHANGES_REQUESTED** [review]
Thanks for working on this @parsharma! It looks like there are a few more places where the dependency on wildfly-elytron should be removed:

* pom.xml
* core-feature-pack/common/pom.xml
* elytron/pom.xml

**Comment by parsharma** [comment]
@fjuma I have not removed elytron dependency from parent/pom.xml because of transitive dependencies.

**Comment by fjuma** [comment]
Thanks for the update @parsharma! To be able to remove the `org.wildfly.security:wildfly-elytron` dependency from the main `pom.xml`, I think we will also need to add an exclusion to the `org.jboss.remoting:jboss-remoting` and the `org.jboss.remotingjmx:remoting-jmx` dependencies in the main `pom.xml` and then work through the compilation errors that occur. From a quick look at the initial compilation failures that will result, it appears that the `domain-management/pom.xml` is missing dependencies on individual Elytron modules.

**Comment by fjuma** [comment]
Thanks very much for the updates @parsharma! The WildFly Core testsuite is now passing - that's great! The WildFly full testsuite is now failing because the `org.wildfly.security:wildfly-elytron` dependency is no longer available. To get that to pass, the next step would be to work on [WFLY-14377](https://issues.redhat.com/browse/WFLY-14377) which involves updating WildFly to depend on the individual Elytron modules instead of `wildfly-elytron`.
 
We have a couple options on how to proceed:

Option 1: Work on and then merge the changes for WFLY-14377 first in order to get CI to pass for this PR.
Option 2: Now that we know all the missing dependencies on individual Elytron modules have been added where needed in this PR, we could temporarily add back the `org.wildfly.security:wildfly-elytronwildfly-elytron` dependency to the main pom.xml here to get CI passing for this PR in order to merge it. We could then work on the changes for WFLY-14377 and then finally remove the `wildfly-elytron` dependency from the main pom.xml for WildFly Core after that.

@darranl Which option do you prefer?

**Comment by darranl** [comment]
Thanks for the ping @fjuma 

Something I am going to be starting some discussions on separately is how we can cut down on code going stale in long running PRs - so on that basis I think option 2, get this merged and then we come back for the final clean up.  As this PR has touched a number of poms there is a very serious risk other PRs will introduce conflicts if this sits too long.  Git is strong in many places, concurrent modifications to XML files is not one of those strengths!

We have a lot of other clean up work that is likely to be iterative anyway so this will not be the first.

Thank you for all the contributions @parsharma this is really good to have WildFly Core tackled.

**Comment by fjuma** [comment]
+1

I've created [WFCORE-5374](https://issues.redhat.com/browse/WFCORE-5374) to track removing the `wildfly-elytron` dependency from the main pom.xml.

**Comment by darranl** [comment]
@parsharma are we ready to take the hold label off this one?  I assume once CI passes we are ready for final review and merge.

**Review by fjuma — CHANGES_REQUESTED** [review]
Just have one small comment. The rest looks good to me. Thanks @parsharma!

**Review by fjuma — APPROVED** [review]
Thanks @parsharma!

**Pull request merged**